script.js

```diff
@@ -0,0 +1,429 @@
+    if (e.target.classList[0] === "in-menu--plus") {
+      const targetItem = e.target.parentNode.parentNode.parentNode;
+      const contentList = document.querySelector(".product-list");
+
+      /*<div class="selected-item-quantity">
+            <div class="add-remove-1">+</div>
+            <div class="add-remove-1">${qty}</div>
+            <div class="add-remove-1">-</div> */
+
+      const index = getIndexInMenu(targetItem, contentList);
+      console.log(index + "hikee");
+
+      const { isVeg, dishName, price } = state.activeMenuList[index];
+
+      for (let i = 0; i < state.cart.length; i++) {
+        if (state.cart[i].dishName === dishName) {
+          console.log(i + "hisss");
+          const updatedState = changeState(state, "increase_in_cart", i);
+          state = updatedState;
+        }
+      }
+    } else {
+      const cartItem = e.target.parentNode.parentNode;
+      const cartList = document.querySelector(".selected-item");
+      const idx = getIndexInMenu(cartItem, cartList);
+
+      const updatedState = changeState(state, "increase_in_cart", idx);
+      state = updatedState;
+    }
+    render();
+  }
+  function decreaseCount(e) {
+    if (e.target.classList[0] === "in-menu--minus") {
+      const targetItem = e.target.parentNode.parentNode.parentNode;
+      const contentList = document.querySelector(".product-list");
+
+      /*<div class="selected-item-quantity">
+            <div class="add-remove-1">+</div>
+            <div class="add-remove-1">${qty}</div>
+            <div class="add-remove-1">-</div> */
+
+      const index = getIndexInMenu(targetItem, contentList);
+
+      const { isVeg, dishName, price } = state.activeMenuList[index];
+
+      for (let i = 0; i < state.cart.length; i++) {
+        if (state.cart[i].dishName === dishName) {
+          const updatedState = changeState(state, "decrease_in_cart", i);
+          state = updatedState;
+        }
+      }
+    } else {
+      const cartItem = e.target.parentNode.parentNode;
+      const cartList = document.querySelector(".selected-item");
+      const idx = getIndexInMenu(cartItem, cartList);
+
+      const updatedState = changeState(state, "decrease_in_cart", idx);
+      state = updatedState;
+    }
+    render();
+  }
+  function handleSearchBar(e) {
+    const keyword = e.target.value.trim().toLowerCase();
+
+    const updatedState = changeState(state, "search_in_bar", {
+      keyword,
+    });
+    state = updatedState;
+    render();
+  }
+  function handleVegFilter(e) {
+    const vegCheckBox = e.target;
+    if (vegCheckBox.checked) {
+      const updatedState = changeState(state, "set_veg_filter");
+      state = updatedState;
+    } else {
+      const updatedState = changeState(state, "remove_veg_filter");
+      state = updatedState;
+    }
+    render();
+  }
+  function handlePromise() {
+    return new Promise(function (resolve, reject) {
+      const userdata = fetch("https://api.github.com/users").then(
```

Comment:
where is this used?

Reply:
Promise function made to handle the API call wherein the function resolve or reject function get execute when the API call is a success or get failed